src/main/kotlin/com/londogard/summarize/extensions/ArrayHelper.kt

```diff
@@ -2,22 +2,41 @@ package com.londogard.summarize.extensions
 
 import kotlin.math.sqrt
 
-internal operator fun Array<Float>.minus(other: Array<Float>): Array<Float> =
-    Array(kotlin.math.min(this.size, other.size)) { i -> this[i] - other[i] }
+internal operator fun Array<Float>.minus(other: Array<Float>): Array<Float> {
+    if (size != other.size)
+        throw IllegalArgumentException("Arrays must be of same size, $size != ${other.size}")
+    for (i in 0 until size) this[i] -= other[i]
 
-internal operator fun Array<Float>.plus(other: Array<Float>): Array<Float> =
-    Array(kotlin.math.min(this.size, other.size)) { i -> this[i] + other[i] }
-
-internal fun Array<Float>.dot(other: Array<Float>): Double =
-    this.zip(other).map { it.first * it.second }.reduce { a, b -> a + b }.toDouble()
+    return this
+}
 
-internal fun Array<Float>.normalize(): Array<Float> {
-    val norm = if (isNotEmpty()) sqrt(map { it * it }.sum()) else 1f
-    for (i in 0 until size) this[i] /= norm
+internal operator fun Array<Float>.plus(other: Array<Float>): Array<Float> {
+    if (size != other.size)
+        throw IllegalArgumentException("Arrays must be of same size, $size != ${other.size}")
+    for (i in 0 until size) this[i] += other[i]
 
     return this
 }
```

Comment:
Catchar vi detta?

Reply:
Och alla de liknande exemplena då ofc

Reply:
Hmm... You're right, I actually made these `internal` so someone using the library can't use the functions. Asserts can be removed.